src/pal/src/init/pal.cpp

```diff
@@ -355,6 +355,39 @@ Initialize(
         gPID = getpid();
         gSID = getsid(gPID);
 
+#ifdef __APPLE__
+        // Store application group Id. It will be null if not set
+        gApplicationGroupId = getenv("NETCOREAPP_SANDBOX_APPLICATION_GROUP_ID");
```

Comment:
@jkotas, I see we've used the "COMPlus_" prefix for other environment variables accessed in the pal.  Do we want to continue using that here?  If not, we've used "DOTNET_" elsewhere, and I'm not aware of anywhere we've used "NETCOREAPP_", so I'd suggest "DOTNET_" as the prefix.